nucypher/blockchain/eth/sol/source/contracts/Issuer.sol

```diff
@@ -114,12 +114,12 @@ contract Issuer is Upgradeable {
     )
         internal returns (uint256 amount)
     {
+        if (currentSupply1 == totalSupply || currentSupply2 == totalSupply) {
+            return;
+        }
         uint256 currentSupply = _currentPeriod <= currentMintingPeriod ?
             Math.min256(currentSupply1, currentSupply2) :
             Math.max256(currentSupply1, currentSupply2);
-        if (currentSupply == totalSupply) {
-            return;
-        }
 
         //totalSupply * lockedValue * (k1 + allLockedPeriods) / (totalLockedValue * k2) -
         //currentSupply * lockedValue * (k1 + allLockedPeriods) / (totalLockedValue * k2)
```

Comment:
this simplifying will decrease accuracy because  there is only integer division in solidity

Reply:
Right now I'm not sure, it was a year ago, I remember that when I simulated function - this equation had better values when `totalSupply` ~ `currentSupply`

Reply:
OK, what I'm saying is that `amount` in L130 can be changed to:

```solidity
amount = totalSupply.sub(currentSupply).mul(_lockedValue).mul(allLockedPeriods).div(denominator);
```

Reply:
I understood, my previous simulation shows that better to have subtraction last, maybe @michwill will help

Reply:
Well, technically both are correct, but the only question is indeed the precision. I don't see why David's variant would perform worse, but probably need to check experimentally

Reply:
I think that @cygnusv's point is (or might be): if you reduce the number of divisions from 2 to 1, with the only division occurring in a case where the denominator is uint256 as here, how is it possible that you can lose precision by dint of being restricted to integer division?  Won't you always increase precision because you are rounding once instead of twice?

Reply:
Additionally, it's more readable in David's variant, yes. And yes, in principle it should be more precise, I guess

Reply:
First of all I don't want any changes in formula in this PR. Also it's was first my variant. I changed it because of simulation end of the curve. To return to this part need to simulate again

Reply:
I think it's ok to leave some things as open issues. I'm this case, since it requires simulation, it makes sense. I'll create an issue.

Reply:
See issue #642. This conversation can be resolved for the moment 👍